### DISCUSSION TITLE
Why not use the database instead of json and yaml?

### DISCUSSION THREAD

**Discussion by bestrocker221** [photoprism/photoprism#3285] [General]
Hi, thanks for the amazing work.
Why not use the database instead of JSON and yaml and any other non-image-video related files?

This way it would avoid creating thousands/millions of files in the file system and rather have everything in the same database?

Are there performance implications?
The database is already in use.
Caching can also be added, either within the DB or with a redis instance.

Having a medium-large library makes up for >1.000.000 files of which ~20% are YML/JSON files

Just trying to understand

**Comment by lastzero** [comment]
You can disable JSON and YAML if you don't want them. But why store JSON in Redis when you can also store it in the file system? It just complicates things and I don't see how this is faster or uses less storage in practice.

**Reply by lastzero** [reply]
See also: https://docs.photoprism.app/user-guide/backups/export/

**Reply by bestrocker221** [reply]
Thank you for your answer, I did not read that page and it explains it.
>Keep in mind that the data still resides in your database, and any changes made to the files will not be reflected in PhotoPrism. This is comparable to doing a backup, but to a human readable format.

I was missing this point. 

Thank you.

**Reply by alexisLefebvre** [reply]
@bestrocker221 did you enable file backups in the past or was it enabled by default?

**Reply by lastzero** [reply]
The config options we provide disable the features, so the default is enabled.

**Reply by alexisLefebvre** [reply]
Can we change it to "disabled" by default?

**Reply by lastzero** [reply]
We can't change the config options as that would break existing behavior. However, we can add this to our config examples if that's not the case already and you believe it should get more attention. We don't set all config options there in order not to make the config file unreadable.

**Reply by bestrocker221** [reply]
Is there a way from the UI to delete all the generated backup yaml and json files ? Or I can just delete them manually?

**Reply by lastzero** [reply]
As with basically everything except the mariadb database files and the static photoprism assets, you can delete them manually.

**Reply by lastzero** [reply]
Note the JSON files are used for indexing Videos and other not directly supported metadata in files, so you will lose or break functionality. See our docs.

**Reply by lastzero** [reply]
This, by the way, is also why we have not disabled them by default.

**Reply by alexisLefebvre** [reply]
Yes the docs explains this: https://docs.photoprism.app/getting-started/faq/#json

So the JSON are necessary but the YAML are not.

**Reply by lastzero** [reply]
- https://github.com/photoprism/photoprism/issues/3320#issuecomment-1493152547

👉 This user says they are a "life saver"
